### ISSUE TITLE
[06-RNN line:55]

### ISSUE THREAD

**Issue by RangoHU** [yunjey/pytorch-tutorial#29]
Hello,

Thanks for the nice tutorial. I have a question, which confuses me a lot, but NOT an issue of your codes. 

In [06-RNN line:55], you compute the output of RNN as "out = self.fc(out[:, -1, :]) ". In other words, you take the hidden values of the last element of all sequences. 

My question is: if we need to take different element from each sequence, i.e. take the hidden values from different positions of the sequences, can the backpropagation work correctly?

For example:

x = Variable(torch.randn(3, 5, 6))
lstm = nn.LSTM(6, 3, batch_first=True)
fc = nn.Linear(3, 1)
a, (b, c) = lstm(x) # a is a 3 x 5 x 3 tensor
out = torch.stack([a[i, j] for i, j in enumerate([1,2,1])]) 
out = fc(out)

criterion = nn.MSELoss()
loss = criterion(out, targets) #targets is a 3x1 tensor

does loss.backward() computes the right grad for all parameters of fc and lstm? 

Thank you so much!

**Comment by RangoHU** [comment]
I am gonna answer my own question... Yes, loss.backward() computes the right grads for all parameters, no matter how modify the inputs :)
